src/pages/JobList/JobList.js

```diff
@@ -1,74 +1,87 @@
 import React from 'react'
-import { PropTypes } from '@dhis2/prop-types'
 import { Card, Checkbox, InputField, IconInfo16 } from '@dhis2/ui'
 import i18n from '@dhis2/d2-i18n'
+import { useJobSchedules } from '../../hooks/job-schedules'
+import { useJobFilter, useShowSystemJobs } from '../../components/Store'
 import { JobTable } from '../../components/JobTable'
 import { LinkButton } from '../../components/LinkButton'
+import { Spinner } from '../../components/Spinner'
 import styles from './JobList.module.css'
 
 const infoLink =
     'https://docs.dhis2.org/en/use/user-guides/dhis-core-version-236/maintaining-the-system/scheduling.html'
 
-const JobList = ({
-    jobs,
-    showSystemJobs,
-    setShowSystemJobs,
-    jobFilter,
-    setJobFilter,
-}) => (
-    <React.Fragment>
-        <header className={styles.header}>
-            <h1 className={styles.headerTitle}>{i18n.t('Scheduled jobs')}</h1>
-            <a
-                href={infoLink}
-                className={styles.headerLink}
-                target="_blank"
-                rel="noopener noreferrer"
-            >
-                <span className={styles.headerLinkIcon}>
-                    <IconInfo16 />
-                </span>
-                {i18n.t('About job configuration')}
-            </a>
-        </header>
-        <Card>
-            <div className={styles.controlContainer}>
-                <InputField
-                    dataTest="job-filter-input"
-                    label={i18n.t('Filter jobs')}
-                    onChange={({ value }) => {
-                        setJobFilter(value)
-                    }}
-                    value={jobFilter}
-                    type="search"
-                    role="searchbox"
-                    name="job-filter"
-                />
-                <div className={styles.controlRight}>
-                    <Checkbox
-                        dataTest="job-toggle-checkbox"
-                        checked={showSystemJobs}
-                        label={i18n.t('Include system jobs in list')}
-                        onChange={({ checked }) => {
-                            setShowSystemJobs(checked)
+const JobList = () => {
+    const [jobFilter, setJobFilter] = useJobFilter()
+    const [showSystemJobs, setShowSystemJobs] = useShowSystemJobs()
+    const { data, loading, error, refetch } = useJobSchedules()
+
+    if (loading) {
+        return <Spinner />
+    }
+
+    if (error) {
+        throw error
```

Comment:
Why not displaying the error like in other components?

Reply:
Yeah you're right. This is something I still need to address 👍

Reply:
Fixed in [890a009](https://github.com/dhis2/scheduler-app/pull/495/commits/890a009186b7806a650f6568a21c38a9174107ca)